File: dofun/safe/deviceinfo/utils/SocUtils.java
package com.dofun.safe.deviceinfo.utils;

import android.app.ActivityManager;
import android.content.Context;
import android.content.pm.ConfigurationInfo;
import android.text.TextUtils;
import androidx.core.util.Pair;
import com.dofun.safe.deviceinfo.bean.CpuBean;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.HashSet;
import java.util.List;

/* loaded from: classes6.dex */
public class SocUtils {
    public static String getSocInfo() {
        String socStr = CommandUtils.execute("getprop ro.board.platform");
        if (!TextUtils.isEmpty(socStr)) {
            return socStr;
        }
        String socStr2 = CommandUtils.execute("getprop ro.hardware");
        if (TextUtils.isEmpty(socStr2)) {
            return CommandUtils.execute("getprop ro.boot.hardware");
        }
        return socStr2;
    }

    public static void setCpuInfo(List<Pair<String, String>> list) {
        try {
            BufferedReader bufferedReader = new BufferedReader(new FileReader("/proc/cpuinfo"));
            CpuBean bean = new CpuBean();
            HashSet<String> parts = new HashSet<>();
            HashSet<String> implementer = new HashSet<>();
            while (true) {
                String line = bufferedReader.readLine();
                if (line != null) {
                    String result = line.toLowerCase();
                    FileLogger.d("soc-", "CPU: " + result);
                    String[] split = result.split(":\\s+", 2);
                    if (split[0].startsWith("cpu part")) {
                        parts.add(split[1]);
                    } else if (split[0].startsWith("hardware")) {
                        bean.setHardware(split[1]);
                    } else if (split[0].startsWith("features")) {
                        bean.setFeatures(split[1]);
                    } else if (split[0].startsWith("cpu implementer")) {
                        implementer.add(split[1]);
                    }
                } else {
                    bean.setParts((String[]) parts.toArray(new String[0]));
                    bean.setImplementers((String[]) implementer.toArray(new String[0]));
                    list.add(new Pair<>("parts", parts.toString()));
                    list.add(new Pair<>("implementer", implementer.toString()));
                    list.add(new Pair<>("hardware", bean.getHardware()));
                    list.add(new Pair<>("features", bean.getFeatures()));
                    return;
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void getGPUInfo(Context context, List<Pair<String, String>> list) {
        ActivityManager am = (ActivityManager) context.getApplicationContext().getSystemService("activity");
        ConfigurationInfo info = am.getDeviceConfigurationInfo();
        list.add(new Pair<>("GlEsVersion", info.getGlEsVersion()));
        list.add(new Pair<>("reqGlEsVersion", info.reqGlEsVersion + ""));
        list.add(new Pair<>("reqInputFeatures", info.reqInputFeatures + ""));
        list.add(new Pair<>("reqKeyboardType", info.reqKeyboardType + ""));
        list.add(new Pair<>("reqNavigation", info.reqNavigation + ""));
        list.add(new Pair<>("reqTouchScreen", info.reqTouchScreen + ""));
        list.add(new Pair<>("describeContents", info.describeContents() + ""));
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/utils/SocUtils.kt
package com.dofun.safe.deviceinfo.utils

import android.app.ActivityManager
import android.content.Context
import android.content.pm.ConfigurationInfo
import android.text.TextUtils
import androidx.core.util.Pair
import com.dofun.safe.deviceinfo.bean.CpuBean
import com.dofun.safe.deviceinfo.utils.Constants
import java.io.BufferedReader
import java.io.FileReader
import java.io.IOException
import java.util.HashSet

object SocUtils {
    fun getSocInfo(): String? {
        val socStr = CommandUtils.execute("getprop ro.board.platform")
        if (!TextUtils.isEmpty(socStr)) {
            return socStr
        }
        val socStr2 = CommandUtils.execute("getprop ro.hardware")
        return if (TextUtils.isEmpty(socStr2)) {
            CommandUtils.execute("getprop ro.boot.hardware")
        } else {
            socStr2
        }
    }

    fun setCpuInfo(list: MutableList<Pair<String, String>>) {
        try {
            val bufferedReader = BufferedReader(FileReader("/proc/cpuinfo"))
            val bean = CpuBean()
            val parts = HashSet<String>()
            val implementer = HashSet<String>()
            while (true) {
                val line = bufferedReader.readLine() ?: break
                val result = line.lowercase()
                FileLogger.d("soc-", "CPU: $result")
                val split = result.split(":\\s+".toRegex(), limit = 2)
                if (split[0].startsWith("cpu part")) {
                    parts.add(split[1])
                } else if (split[0].startsWith("hardware")) {
                    bean.hardware = split[1]
                } else if (split[0].startsWith("features")) {
                    bean.features = split[1]
                } else if (split[0].startsWith("cpu implementer")) {
                    implementer.add(split[1])
                }
            }
            bean.parts = parts.toTypedArray()
            bean.implementers = implementer.toTypedArray()
            list.add(Pair("parts", parts.toString()))
            list.add(Pair("implementer", implementer.toString()))
            list.add(Pair("hardware", bean.hardware ?: Constants.UNKNOWN))
            list.add(Pair("features", bean.features ?: Constants.UNKNOWN))
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun getGPUInfo(context: Context, list: MutableList<Pair<String, String>>) {
        val am = context.applicationContext.getSystemService("activity") as ActivityManager
        val info: ConfigurationInfo = am.deviceConfigurationInfo
        list.add(Pair("GlEsVersion", info.glEsVersion))
        list.add(Pair("reqGlEsVersion", info.reqGlEsVersion.toString()))
        list.add(Pair("reqInputFeatures", info.reqInputFeatures.toString()))
        list.add(Pair("reqKeyboardType", info.reqKeyboardType.toString()))
        list.add(Pair("reqNavigation", info.reqNavigation.toString()))
        list.add(Pair("reqTouchScreen", info.reqTouchScreen.toString()))
        list.add(Pair("describeContents", info.describeContents().toString()))
    }
}
